### PR TITLE
feat(gas_service)!: pay gas from ticket

### PR DESCRIPTION
**Pull request by Foivos**
[AXE-6658]

[AXE-6658]: https://axelarnetwork.atlassian.net/browse/AXE-6658?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: move/gas_service/sources/versioned/gas_service_v0.move [hunk 1]
```diff
@@ -31,44 +32,43 @@ public(package) fun version_control(self: &GasService_v0): &VersionControl {
...
+    events::gas_added<SUI>(
+        message_id,
         coin_value,
         refund_address,
         params,
     );
 }
 
-public(package) fun add_gas(
+public(package) fun pay_gas(
```
Comment (milapsheth) [inline move/gas_service/sources/versioned/gas_service_v0.move:53]: move pay_gas above add_gas for clarity

File: move/gas_service/sources/gas_service.move [hunk 2]
```diff
@@ -64,49 +65,43 @@ macro fun value_mut(
...
-/// Add gas for an existing cross-chain contract call.
-/// This function can be called by a user who wants to add gas for a contract
-/// call with insufficient gas.
-public fun add_gas(
+/// Pay gas for a contract call.
+/// This function is called by the channel that wants to pay gas for a contract
+/// call.
+/// It can also be called by the user to pay gas for a contract call, while
+/// setting the sender as the channel ID.
+public fun pay_gas(
```
Comment (milapsheth) [inline move/gas_service/sources/gas_service.move:93]: move pay_gas above add_gas for clarity

File: .changeset/fair-dots-explode.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@axelar-network/axelar-cgp-sui': minor
+---
+
+Paying for gas now requires the message ticket instead of the call information.
```
Comment (milapsheth) [inline .changeset/fair-dots-explode.md:5]: check if sui deployment/operational scripts need to be updated

**Review by milapsheth — APPROVED** [review]

File: move/gas_service/sources/gas_service.move (hunk 2 above)
Comment (Foivos) [reply move/gas_service/sources/gas_service.move]: done

File: .changeset/fair-dots-explode.md (hunk 3 above)
Comment (Foivos) [reply .changeset/fair-dots-explode.md:5]: it does not

File: move/gas_service/sources/versioned/gas_service_v0.move (hunk 1 above)
Comment (Foivos) [reply move/gas_service/sources/versioned/gas_service_v0.move]: done

**Review by milapsheth — APPROVED** [review]

**Pull request merged**